packages/d/discord/xmake.lua

```diff
@@ -4,26 +4,54 @@ package("discord")
+    add_patches("3.2.1", path.join(os.scriptdir(), "patches", "3.2.1", "add_include_cstdint_to_typeh.patch"), "0ae6618dd5bf2e0149bbb3959dcd2f6df5b2a8e7295b7153eea2fd1e6d389ba0")
 
     set_urls("https://dl-game-sdk.discordapp.net/$(version)/discord_game_sdk.zip")
 
-    on_install("windows", function (package)
-        os.cp("cpp/*.h", package:installdir("include"))
-        os.cp("c/*.h", package:installdir("include"))
-        os.cp("Release/*.lib", package:installdir("lib"))
-        if package:is_arch("x64") then
-            os.cp("lib/x86_64/discord_game_sdk.dll", package:installdir("bin"))
-            os.cp("lib/x86_64/discord_game_sdk.dll.lib", package:installdir("lib"))
-        else
-            os.cp("lib/x86/discord_game_sdk.dll", package:installdir("bin"))
-            os.cp("lib/x86/discord_game_sdk.dll.lib", package:installdir("lib"))
+    add_configs("shared", {description = "Use shared binaries.", default = false, type = "boolean", readonly = true})
+    add_configs("cppapi", {description = "Enable C++ API.", default = true, type = "boolean"})
+
+    on_install("windows|x86", "windows|x64", "linux|x64", "macosx|x86_64", "macosx|arm64", function (package)
+        if package:configs("cppapi") then
+            os.cp("cpp/*.h", package:installdir("include"))
         end
+        os.cp("c/*.h", package:installdir("include"))
 
         local configs = {}
-        if package:config("shared") then
-            configs.kind = "shared"
+        if package:is_plat("windows") then
+            if package:is_arch("x64") then
+                os.cp("lib/x86_64/discord_game_sdk.dll", package:installdir("bin"))
+                os.cp("lib/x86_64/discord_game_sdk.dll.lib", package:installdir("lib"))
+                configs.linkdirs = "lib/x86_64"
+            else
+                os.cp("lib/x86/discord_game_sdk.dll", package:installdir("bin"))
+                os.cp("lib/x86/discord_game_sdk.dll.lib", package:installdir("lib"))
+                configs.linkdirs = "lib/x86"
+            end
+            configs.links = "discord_game_sdk.dll"
+        elseif package:is_plat("linux") then
+                os.cp("lib/x86_64/discord_game_sdk.so", package:installdir("lib"))
+                configs.linkdirs = "lib/x86_64"
+                configs.links = "discord_game_sdk"
+        elseif package:is_plat("macosx") then
+            local version = package:version()
+            if ((version:major() > 3) or (version:major() == 3 and version:minor() >= 2)) and package:is_arch("arm64") then
+                os.cp("lib/aarch64/discord_game_sdk.dylib", package:installdir("lib"))
+                configs.linkdirs = "lib/aarch64"
+
+            else
+                os.cp("lib/x86_64/discord_game_sdk.dylib", package:installdir("lib"))
+                configs.linkdirs = "lib/x86_64"
+            end
+            configs.links = "discord_game_sdk.dylib"
```

Comment:
I don't understand why links are set to discordcpp when it is a static library.

Shouldn't `add_links("discordcpp", "discord_game_sdk")` or package:add("discordcpp", "discord_game_sdk") be configured in the package and then passed to the user project?